File: app/src/main/java/dev/ragnarok/fenrir/fragment/search/options/SimpleTextOption.java
package dev.ragnarok.fenrir.fragment.search.options;

import android.os.Parcel;

import androidx.annotation.NonNull;

import java.util.Objects;

public class SimpleTextOption extends BaseOption {

    public static final Creator<SimpleTextOption> CREATOR = new Creator<SimpleTextOption>() {
        @Override
        public SimpleTextOption createFromParcel(Parcel in) {
            return new SimpleTextOption(in);
        }

        @Override
        public SimpleTextOption[] newArray(int size) {
            return new SimpleTextOption[size];
        }
    };
    public String value;

    public SimpleTextOption(int key, int title, boolean active) {
        super(SIMPLE_TEXT, key, title, active);
    }

    protected SimpleTextOption(Parcel in) {
        super(in);
        value = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
        dest.writeString(value);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;

        SimpleTextOption that = (SimpleTextOption) o;
        return Objects.equals(value, that.value);
    }

    @Override
    public int hashCode() {
        int result = super.hashCode();
        result = 31 * result + (value != null ? value.hashCode() : 0);
        return result;
    }

    @NonNull
    @Override
    public SimpleTextOption clone() throws CloneNotSupportedException {
        SimpleTextOption clone = (SimpleTextOption) super.clone();
        clone.value = value;
        return clone;
    }

    @Override
    public int describeContents() {
        return 0;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/fragment/search/options/SimpleTextOption.kt
package dev.ragnarok.fenrir.fragment.search.options

import android.os.Parcel
import android.os.Parcelable

class SimpleTextOption : BaseOption {
    @JvmField
    var value: String? = null

    constructor(key: Int, title: Int, active: Boolean) : super(SIMPLE_TEXT, key, title, active)
    private constructor(`in`: Parcel) : super(`in`) {
        value = `in`.readString()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
        dest.writeString(value)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        if (!super.equals(other)) return false
        val that = other as SimpleTextOption
        return value == that.value
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + if (value != null) value.hashCode() else 0
        return result
    }

    @Throws(CloneNotSupportedException::class)
    override fun clone(): SimpleTextOption {
        val clone = super.clone() as SimpleTextOption
        clone.value = value
        return clone
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SimpleTextOption> {
        override fun createFromParcel(parcel: Parcel): SimpleTextOption {
            return SimpleTextOption(parcel)
        }

        override fun newArray(size: Int): Array<SimpleTextOption?> {
            return arrayOfNulls(size)
        }
    }
}